openeo/metadata.py

```diff
@@ -540,6 +540,22 @@ def metadata_from_stac(url: str) -> CubeMetadata:
 
     # TODO move these nested functions and other logic to _StacMetadataParser
 
+    def get_temporal_metadata(spec: dict, complain: Callable[[str], None] = warnings.warn) -> TemporalDimension:
+        # Dimension info is in `cube:dimensions` (or 0.4-style `properties/cube:dimensions`)
+        cube_dimensions = (
+            deep_get(spec, "cube:dimensions", default=None)
+            or deep_get(spec, "properties", "cube:dimensions", default=None)
+            or {}
+        )
+        if not cube_dimensions:
+            complain("No cube:dimensions metadata")
+        dimensions = []
+        for name, info in cube_dimensions.items():
+            dim_type = info.get("type")
+            if dim_type == "temporal":
+                return TemporalDimension(name=name, extent=info.get("extent"))
+        return []
```

Comment:
returning an empty list as fallback looks weird. Shouldn't this be return `None` instead?